File: Tutorials/Tuckshop/Client/Tuckshop.WebApp/src/Domain/Models/Wallets/Commands/DepositToWallet.ts
import { Attributes, ModelBase, Rules, Validation } from '@singularsystems/neo-core';
export default class DepositToWallet extends ModelBase {
    static typeName = "DepositToWallet";

    constructor() {
        super();
        this.makeObservable();
    }

    @Rules.Required()
    public customerId: number = 0;

    @Attributes.Float()
    public amount: number = 0;

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<DepositToWallet>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew) {
            return "New deposit to wallet";
        } else {
            return "Deposit To Wallet";
        }
    }
}

File: Tutorials/Tuckshop/Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Wallets/Commands/DepositToWallet.cs
namespace Tuckshop.Core.Models.Wallets.Commands
{
  using Neo.Model;
  using System.ComponentModel.DataAnnotations;
  using System.ComponentModel.DataAnnotations.Schema;

  /// <summary>
  /// Class representing a deposit to a customer's wallet.
  /// </summary>
  public class DepositToWallet : ModelBase<DepositToWallet>
  {
    /// <summary>
    /// Gets and sets the Customer Id.
    /// </summary>
    [Required]
    public int CustomerId { get; set; }

    /// <summary>
    /// Gets and sets the Amount to deposit.
    /// </summary>
    [Column(TypeName = "money")]
    public decimal Amount { get; set; }

    //// <inheritdoc/>
    //protected override void AddBusinessRules(ValidationRules<DepositToWallet> rules)
    //{
    //  rules.FailWhen(c => c.Amount <= 0, "Deposit amount must be greater than zero.");
    //}
  }
}